app/src/main/java/org/mozilla/fenix/perf/ProfilerUtils.kt

```diff
@@ -0,0 +1,150 @@
+import mozilla.components.concept.fetch.Response
+import org.json.JSONObject
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+
+private const val PROFILER_API = "https://api.profiler.firefox.com/compressed-store"
+private const val PROFILER_SERVER_HEADER = "application/vnd.firefox-profiler+json;version=1.0"
+private const val TOKEN = "profileToken"
+private const val PROFILER_DATA_URL = "https://profiler.firefox.com/public/"
+
+private val firefox_features = arrayOf(
+    "screenshots", "js", "leaf", "stackwalk", "cpu", "java",
+    "processcpu", "ipcmessages", "java"
+)
+private val firefox_threads = arrayOf(
+    "GeckoMain", "Compositor", "Renderer",
+    "SwComposite", "DOM Worker"
+)
+
+private val graphics_features = arrayOf("java", "ipcmessages")
+private val graphics_threads = arrayOf(
+    "GeckoMain", "Compositor", "Renderer", "SwComposite",
+    "RenderBackend", "SceneBuilder", "WrWorker", "CanvasWorkers",
+)
+
+private val media_features = arrayOf(
+    "js", "leaf", "stackwalk", "cpu", "audiocallbacktracing",
+    "ipcmessages", "processcpu", "java"
+)
+private val media_threads = arrayOf(
+    "cubeb", "audio", "camera", "capture", "Compositor", "GeckoMain", "gmp", "graph", "grph",
+    "InotifyEventThread", "IPDL Background", "media", "ModuleProcessThread", "PacerThread",
+    "RemVidChild", "RenderBackend", "Renderer", "Socket Thread", "SwComposite",
+    "webrtc"
+)
+
+private val networking_features = arrayOf(
+    "screenshots", "js", "leaf", "stackwalk", "cpu", "java",
+    "processcpu", "ipcmessages"
+)
+
+private val networking_threads = arrayOf(
+    "Compositor", "DNS Resolver", "DOM Worker", "GeckoMain",
+    "Renderer", "Socket Thread", "StreamTrans", "SwComposite", "TRR Background"
+)
+
+/**
+ * Profiler settings enum for grouping features and settings together
+ */
+enum class ProfilerSettings(val threads: Array<String>, val features: Array<String>) {
+    Firefox(firefox_threads, firefox_features),
+    Graphics(graphics_threads, graphics_features),
+    Media(media_threads, media_features),
+    Networking(networking_threads, networking_features);
+}
+
+/**
+ * Collection of functionality to parse and save the profile returned by GeckoView.
+ */
+object ProfilerUtils {
+
+    private fun saveProfileUrlToClipboardAndToast(profileResult: ByteArray, context: Context): String {
+        // The profile is saved to a temporary file since our fetch API takes a file or a string.
+        // Converting the ByteArray to a String would hurt the encoding, which we need to preserve.
+        val outputFile = createTemporaryFile(profileResult, context)
+        try {
+            val response = networkCallToProfilerServer(outputFile, context)
+            val profileToken = decodeProfileToken(response)
+            return PROFILER_DATA_URL + profileToken
+        } finally {
+            outputFile.delete()
```

Comment:
todo (blocking): we should also add a comment mentioning this file contains sensitive information and it should be handled carefully.